dev/io.openliberty.org.apache.myfaces.4.0/src/META-INF/resources/jakarta.faces/faces.js

```diff
@@ -0,0 +1 @@
 B=myfaces._impl.core._Runtime.getGlobalConfig("facesAjaxImpl",myfaces._impl.core.Impl);return B.getSeparatorChar();}};window.faces=faces;}if(!faces.ajax){faces.ajax=new function(){this.request=function(C,D,B){if(!B){B={};}var A=myfaces._impl.core._Runtime.getGlobalConfig("facesAjaxImpl",myfaces._impl.core.Impl);return A.request(C,D,B);};this.addOnError=function(B){var A=myfaces._impl.core._Runtime.getGlobalConfig("facesAjaxImpl",myfaces._impl.core.Impl);return A.addOnError(B);};this.addOnEvent=function(B){var A=myfaces._impl.core._Runtime.getGlobalConfig("facesAjaxImpl",myfaces._impl.core.Impl);return A.addOnEvent(B);};this.response=function(C,B){var A=myfaces._impl.core._Runtime.getGlobalConfig("facesAjaxImpl",myfaces._impl.core.Impl);return A.response(C,B);};};}if(!faces.util){faces.util=new function(){this.chain=function(C,B){var A=myfaces._impl.core._Runtime.getGlobalConfig("facesAjaxImpl",myfaces._impl.core.Impl);return A.chain.apply(A,arguments);};};}if(!faces.push){faces.push=new function(){var C=window.location.protocol.replace("http","ws")+"//";var H=500;var D=25;var K="Expired";var I={};var G={};var F={};var L={};function B(P,O,Q){var M;var R=0;var N=this;N.open=function(){if(M&&M.readyState==1){return ;}M=new WebSocket(O);M.onopen=function(T){if(!R){var U=F[P];for(var S=U.length-1;S>=0;S--){var V=U[S];G[V]["onopen"](Q);}}R=0;};M.onmessage=function(W){var V=JSON.parse(W.data);for(var U=F[P].length-1;U>=0;U--){var Z=F[P][U];if(document.getElementById(Z)){try{G[Z]["onmessage"](V,Q,W);}catch(Y){}var S=G[Z]["behaviors"];var X=S[V];if(X&&X.length){for(var T=0;T<X.length;T++){try{X[T](null);}catch(Y){}}}}else{F[P].splice(U,1);}}if(F[P].length==0){N.close();}};M.onclose=function(T){if(!M||(T.code==1000&&T.reason==K)||(T.code==1008)||(!R)||(R>=D)){var U=F[P];for(var S=U.length-1;S>=0;S--){var V=U[S];G[V]["onclose"](T.code,Q,T);}}else{setTimeout(N.open,H*R++);}};M.onerror=function(T){var U=F[P];for(var S=U.length-1;S>=0;S--){var V=U[S];G[V]["onerror"](Q);}};};N.close=function(){if(M){var S=M;M=null;S.close();}};}this.init=function(N,O,Q,U,P,R,T,S,M){T=J(T);if(!window.WebSocket){T(-1,Q);return ;}var V=O.substr(O.indexOf("?")+1);if(!G[N]){G[N]={"channelToken":V,"onopen":J(U),"onmessage":J(P),"onerror":J(R),"onclose":T,"behaviors":S,"autoconnect":M};if(!F[V]){F[V]=[];}F[V].push(N);if(!I[V]){I[V]=new B(V,A(O),Q);}}if(M){this.open(N);}};this.open=function(M){E(G[M]["channelToken"]).open();};this.close=function(M){E(G[M]["channelToken"]).close();};function A(M){if(M.indexOf("://")<0){var N=window.location.hostname+":"+window.location.port;return C+N+M;}else{return M;}}function E(N){var M=I[N];if(M){return M;}else{throw new Error("Unknown channelToken: "+N);}}function J(M){return(typeof M!=="function")&&(M=window[M]||function(){}),M;}};}(!window.myfaces)?window.myfaces={}:null;if(!myfaces.ab){myfaces.ab=function(F,D,A,E,C,B){if(!B){B={};}if(A){B["jakarta.faces.behavior.event"]=A;}if(E){B["execute"]=E;}if(C){B["render"]=C;}faces.ajax.request(F,D,B);};}
```

Comment:
Is there a way to actually see the diff for this?

Reply:
Not really, as this is just one line of javascript code.  I tried using the source, but encountered other js errors.  I had to use this updated file (after building upstream 4.0)

Reply:
MD5 for this updated file is `ca60b72be475f3d0d9e59a1c77abb6ed` if you'd like to verify, too